plugin/src/server/sc/plugin2021/Game.kt

```diff
@@ -167,6 +169,9 @@ class Game: RoundBasedGameInstance<Player>(GamePlugin.PLUGIN_UUID) {
     override fun getCurrentState(): IGameState = gameState
     
     fun isGameOver(): Boolean {
+        if (round > Constants.ROUND_LIMIT) {
+            gameState.orderedColors.clear()
+        }
         GameRuleLogic.removeInvalidColors(gameState)
         return gameState.orderedColors.isEmpty()
```

Comment:
You need to separate the part out that does these modifications - an `is` method with sideeffects can be quite nasty.
My suggestion: Turn `isGameOver` into a `val` that only does the `isEmpty` check and then create a new method `checkGameOver` that behaves like this method now.

Reply:
Refactored in 25281d2f